src/main/java/org/peergos/client/MerkleNode.java

```diff
@@ -0,0 +1,104 @@
+package org.peergos.client;
+
+import io.ipfs.cid.Cid;
+import io.ipfs.multihash.Multihash;
+import org.peergos.util.JSONParser;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class MerkleNode {
+
+    public final Multihash hash;
+    public final Optional<String> name;
+    public final Optional<Integer> size;
+    public final Optional<String> largeSize;
+    public final Optional<Integer> type;
+    public final List<MerkleNode> links;
+    public final Optional<byte[]> data;
+
+    public MerkleNode(String hash,
```

Comment:
Do we need all this stuff? Can we simplify this as we controls boths ends?

Reply:
done, put() now returns Cid